letsencrypt/plugins/webroot.py

```diff
@@ -157,3 +253,54 @@ def cleanup(self, achalls):  # pylint: disable=missing-docstring
                                      root_path)
                     else:
                         raise
+
+
+class _WebrootMapAction(argparse.Action):
+    """Action class for parsing webroot_map."""
+
+    def __call__(self, parser, namespace, webroot_map, option_string=None):
+        for domains, webroot_path in six.iteritems(json.loads(webroot_map)):
+            validated_webroot_path = _validate_webroot(webroot_path)
+            namespace.webroot_map.update(
+                (d, validated_webroot_path,)
+                for d in cli.add_domains(namespace, domains))
```

Comment:
This multi-line call with a tuple comprehension in it is hard to read. Stylistically, this might be nicer?
```
new = ((d, validated_path) for d in cli.add_domains(namespace, domains))
namespace.webroot_map.update(new)
```